dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -369,6 +369,93 @@ def diff(self, other_profile: NumericStatsMixin, options: Dict = None) -> Dict:
                 "and '{}'".format(cls.__name__, other_profile.__class__.__name__)
             )
 
+        new_self_histogram = {"bin_counts": None, "bin_edges": None}
+        new_other_histogram = {"bin_counts": None, "bin_edges": None}
+        regenerate_histogram = False
+        num_psi_bins = 10
+
+        if (
+            isinstance(self._stored_histogram["histogram"]["bin_counts"], np.ndarray)
+            and isinstance(self._stored_histogram["histogram"]["bin_edges"], np.ndarray)
+            and isinstance(
+                other_profile._stored_histogram["histogram"]["bin_counts"], np.ndarray
+            )
+            and isinstance(
+                other_profile._stored_histogram["histogram"]["bin_edges"], np.ndarray
+            )
+        ):
+            regenerate_histogram = True
+            min_min_edge = min(
+                self._stored_histogram["histogram"]["bin_edges"][0],
+                other_profile._stored_histogram["histogram"]["bin_edges"][0],
+            )
+            max_max_edge = max(
+                self._stored_histogram["histogram"]["bin_edges"][-1],
+                other_profile._stored_histogram["histogram"]["bin_edges"][-1],
+            )
+
+        if regenerate_histogram:
+            len_self_bin_counts = 0
+            if len(self._stored_histogram["histogram"]["bin_counts"]) > 0:
+                len_self_bin_counts = len(
+                    self._stored_histogram["histogram"]["bin_counts"]
+                )
+
+            # re-calculate `self` histogram
+            if not len_self_bin_counts == num_psi_bins:
+                histogram, hist_loss = self._regenerate_histogram(
+                    bin_counts=self._stored_histogram["histogram"]["bin_counts"],
+                    bin_edges=self._stored_histogram["histogram"]["bin_edges"],
+                    suggested_bin_count=num_psi_bins,
+                    options={
+                        "min_edge": min_min_edge,
+                        "max_edge": max_max_edge,
+                    },
+                )
+                new_self_histogram["bin_counts"] = histogram["bin_counts"]
+                new_self_histogram["bin_edges"] = histogram["bin_edges"]
+
+            if len_self_bin_counts == num_psi_bins:
```

Comment:
Instead of this if, could set it above the first if, and it will get reassigned given the criteria

Reply:
totally fair -- updating now